src/fabric.c

```diff
@@ -563,32 +586,50 @@ static void ofi_ini_dir(const char *dir)
 			       "asprintf failed to allocate memory\n");
 			goto libdl_done;
 		}
-		FI_DBG(&core_prov, FI_LOG_CORE, "opening provider lib %s\n", lib);
-
-		dlhandle = dlopen(lib, RTLD_NOW);
+		ofi_reg_dl_prov(lib);
+	
 		free(liblist[n]);
-		if (dlhandle == NULL) {
-			FI_WARN(&core_prov, FI_LOG_CORE,
-			       "dlopen(%s): %s\n", lib, dlerror());
-			free(lib);
-			continue;
-		}
 		free(lib);
-
-		inif = dlsym(dlhandle, "fi_prov_ini");
-		if (inif == NULL) {
-			FI_WARN(&core_prov, FI_LOG_CORE, "dlsym: %s\n", dlerror());
-			dlclose(dlhandle);
-		} else {
-			ofi_register_provider((inif)(), dlhandle);
-		}
 	}
 
 libdl_done:
 	while (n-- > 0)
 		free(liblist[n]);
 	free(liblist);
 }
+
+static void ofi_find_prov_libs(void)
+{
+	const char* lib_prefix = "lib";
+	const char* lib_postfix = "-fi.so";
+	struct ofi_prov *prov;
+
+	/* Try to find the libraries that are visible by their name
+		* (LD_LIBRARY_PATH, system paths) */
```

Comment:
Please fix-up indentation and move comment above the function.  I would change up the comment slightly, e.g.:
```
/* Search standard system library paths (i.e. LD_LIBRARY_PATH) for known DL provider
 * libraries.
 */
```